src/Bicep.Core.Samples/DataSets.cs

```diff
@@ -52,6 +53,13 @@ public static class DataSets
                 .Select(property => property.GetValue(null))
                 .Cast<DataSet>();
 
+        public static IEnumerable<DataSet> DataSetsWithNoDiagnostics =>
+            typeof(DataSets)
+                .GetProperties(BindingFlags.Public | BindingFlags.GetProperty | BindingFlags.Static)
+                .Where(property => property.PropertyType == typeof(DataSet) && !property.Name.StartsWith("Invalid", StringComparison.Ordinal))
```

Comment:
>!property.Name.StartsWith("Invalid", StringComparison.Ordinal) [](start = 79, length = 62)

We do have the `IsValid` property on the `DataSet` class. Perf-wise it shouldn't be noticeable because the DataSets are lazy loaded anyway.

Reply:
Nice! I didn't notice that. Updated the it to use `IsValid`